homeassistant/components/bitcoin/sensor.py

```diff
@@ -25,34 +31,120 @@
-    "blocks_size": ["Block size", None],
-    "total_fees_btc": ["Total fees", "BTC"],
-    "total_btc_sent": ["Total sent", "BTC"],
-    "estimated_btc_sent": ["Estimated sent", "BTC"],
-    "total_btc": ["Total", "BTC"],
-    "total_blocks": ["Total Blocks", None],
-    "next_retarget": ["Next retarget", None],
-    "estimated_transaction_volume_usd": ["Est. Transaction volume", "USD"],
-    "miners_revenue_btc": ["Miners revenue", "BTC"],
-    "market_price_usd": ["Market price", "USD"],
-}
+SENSOR_TYPES: tuple[SensorEntityDescription, ...] = (
+    SensorEntityDescription(
+        key="exchangerate",
+        name="Exchange rate (1 BTC)",
+        unit_of_measurement=None,
+    ),
+    SensorEntityDescription(
+        key="trade_volume_btc",
+        name="Trade volume",
+        unit_of_measurement="BTC",
+    ),
+    SensorEntityDescription(
+        key="miners_revenue_usd",
+        name="Miners revenue",
+        unit_of_measurement="USD",
+    ),
+    SensorEntityDescription(
+        key="btc_mined",
+        name="Mined",
+        unit_of_measurement="BTC",
+    ),
+    SensorEntityDescription(
+        key="trade_volume_usd",
+        name="Trade volume",
+        unit_of_measurement="USD",
+    ),
+    SensorEntityDescription(
+        key="difficulty",
+        name="Difficulty",
+        unit_of_measurement=None,
+    ),
+    SensorEntityDescription(
+        key="minutes_between_blocks",
+        name="Time between Blocks",
+        unit_of_measurement=TIME_MINUTES,
+    ),
+    SensorEntityDescription(
+        key="number_of_transactions",
+        name="No. of Transactions",
+        unit_of_measurement=None,
+    ),
+    SensorEntityDescription(
+        key="hash_rate",
+        name="Hash rate",
+        unit_of_measurement=f"PH/{TIME_SECONDS}",
+    ),
+    SensorEntityDescription(
+        key="timestamp",
+        name="Timestamp",
+        unit_of_measurement=None,
+    ),
+    SensorEntityDescription(
+        key="mined_blocks",
+        name="Mined Blocks",
+        unit_of_measurement=None,
+    ),
+    SensorEntityDescription(
+        key="blocks_size",
+        name="Block size",
+        unit_of_measurement=None,
+    ),
+    SensorEntityDescription(
+        key="total_fees_btc",
+        name="Total fees",
+        unit_of_measurement="BTC",
+    ),
+    SensorEntityDescription(
+        key="total_btc_sent",
+        name="Total sent",
+        unit_of_measurement="BTC",
+    ),
+    SensorEntityDescription(
+        key="estimated_btc_sent",
+        name="Estimated sent",
+        unit_of_measurement="BTC",
+    ),
+    SensorEntityDescription(
+        key="total_btc",
+        name="Total",
+        unit_of_measurement="BTC",
+    ),
+    SensorEntityDescription(
+        key="total_blocks",
+        name="Total Blocks",
+        unit_of_measurement=None,
+    ),
+    SensorEntityDescription(
+        key="next_retarget",
+        name="Next retarget",
+        unit_of_measurement=None,
```

Comment:
`None` is the default `unit_of_measurement`

Reply:
I'm aware of that. The intention here was to keep it uniform. I'm happy to remove the `None` values though if you think that would be better.

Reply:
Personally, I don't think they should be defined. But I see your point :shrug:

Reply:
Please remove the defaults.